plenoptic/synthesize/eigendistortion.py

```diff
@@ -374,170 +386,51 @@ def _synthesize_power(self, l=0, tol=1e-10, n_steps=1000):
 
         return lmbda_new, v_new
 
-    def _clamp_and_store(self, idx, v, d_lambda):
-        """Overwrite base class _clamp_and_store. We don't actually need to clamp the signal."""
-        if self.store_progress:
-            v = self._vector_to_image(v)[0]
-            # self._all_saved_signals[idx].append(v)
-            self._all_losses[idx].append(d_lambda.item())
-
-    def _synthesize_lanczos(self, n_steps=1000, e_vecs=None, debug_A=None):
-        r""" Lanczos Algorithm with full reorthogonalization after each iteration.
-
-        Computes approximate eigenvalues/vectors of FIM (i.e. the Ritz values and vectors). Each vector returned from
-        power iteration is saved (i.e. :math:`v, Av, Avv, Avvv`, etc.). These vectors span the Krylov subspace which is a good
-        approximation of the matrix. Each Lanczos iteration orthogonalizes the current vector against all previously
-        obtained vectors. Matrix :math:`A` is the FIM which is computed implicitly. This orthogonalization is done via
-        Gram-Schmidt orthogonalization of the current vector against all previous vectors at each iteration. Since this
-        procedure is done by projecting the current vector onto all previous vectors (via matrix-vector multiplication),
-        then removing the projected component at each iteration, later iterations to take longer than early iterations.
-
-        This method simultaneously approximates both ends of the eigenspectrum. This is analyzed in depth in the below
-        references [1] and [2].
-
-        We want to estimate :math:`k` eigenvalues and eigenvectors of our Fisher information matrix, :math:`A`. This is done
-        by decomposing :math:`A` as
-
-        .. math::
-            \begin{align}
-            A &\approx  Q_kT_kQ_K^T\\
-            \text{where } T_{k}&=\left[\begin{array}{cccc}{\alpha_{1}} & {\beta_{2}} & {} & {} \\ {\beta_{2}} & {\ddots} & {\ddots} & {} \\ {} & {\ddots} & {\ddots} & {\beta_{k}} \\ {} & {} & {\beta_{k}} & {\alpha_{k}}\end{array}\right]\\
-            Q_k &= [\bf{q_1}, \bf{q_2}, ..., \bf{q_k}].
-            \end{align}
-
-        Here :math:`T_k` is a :math:`(k \times k)` tri-diagonal matrix, and :math:`Q_k` is an :math:`(n\times k)` matrix of
-        orthogonal vectors spanning the k-dimensional Krylov subspace of :math:`A`. The matrix :math:`T_k` is easier to
-        diagonalize due to its reduced size and tridiagonal structure, :math:`T_k=V\Lambda V^T`. By subbing this into our
-        earlier approximation for :math:`A`, we find
-
-        .. math::
-            \begin{align}
-            A &\approx Q_kT_kQ_k^T\\
-              &= Q_kV\Lambda V^T Q_k^T
-            \end{align}
-
-        Thus, :math:`Q_kV` are our estimates of :math:`k` eigenvectors of :math:`A` and :math:`\text{diag}(\Lambda )` are
-        their associated eigenvalues.
-
+    def _synthesize_randomized_svd(self, k, p=0, q=0):
+        r"""  Synthesize eigendistortions using randomized truncated SVD.
```

Comment:
Right now, user cannot set `q` or `p`, and they're hard coded to `p=5`, `q=2`. Are these parameters that users would ever want to change? E.g., are they hyperparameters that will need to be tweaked to find a good solution?

Reply:
These should work 99% of the time, but you're right. I've now allowed the user to change it and just set ``p=5`` ``q=2`` as defaults.